pkg/internal/kubernetes/provider_id_test.go

```diff
@@ -0,0 +1,54 @@
+package kubernetes
+
+import "testing"
+
+func TestParseProvider(t *testing.T) {
+	tests := []struct {
+		name     string
+		value    string
+		id       string
+		provider string
+		error    bool
+	}{
+		{
+			name:  "extra slashes",
+			value: "docker:////my-machine",
+			id:    "my-machine",
+		},
+		{
+			name:  "two slashes",
+			value: "docker://another-machine",
+			id:    "another-machine",
+		},
+		{
+			name:  "no slashes",
+			value: "just-some-value",
+			error: true,
+		},
+		{
+			name:  "amazon with extras",
+			value: "amazon://some/values/machinename",
+			id:    "machinename",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			id, err := ParseProviderID(test.value)
```

Comment:
I read these as 2 different test cases. Trying to shove error conditions into a test case that is only checking successful responses makes the logic in this for loop too complex. These t.Run functions should be as small as possible with as little logic as possible.

The two test cases I see are `TestProviderIDReturnsErrorWithBadInput`, `TestProviderIDReturnsProviderID`